pkg/container/qemu_runner.go

```diff
@@ -498,10 +492,14 @@ func (b qemuOCILoader) RemoveImage(ctx context.Context, ref string) error {
 
 func createMicroVM(ctx context.Context, cfg *Config) error {
 	log := clog.FromContext(ctx)
-	log.Debug("qemu: ssh - create ssh key pair")
-	pubKey, err := generateSSHKeys(ctx, cfg)
-	if err != nil {
-		return err
+	log.Debug("qemu: ssh - get user ssh key pair")
+	pubKey, err := getUserSSHKey(cfg)
+	if err != nil || pubKey == nil {
```

Comment:
IIUC this switches away from ephemeral keys by default and instead uses the user's SSH key pairs, if it can?

I would rather stick to an ephemeral key by default, but maybe we do this by default under `debug`?

Reply:
Right, I've now made so that user's keys are injected just before the debug shell, this way we keep the ephemeral keypair during the whole build until the debug point happens